### PR TITLE
LPS-142887

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related Issue: https://issues.liferay.com/browse/LPS-142887

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:stop

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/util/DDMFormFieldFactoryHelperTest.java [hunk 1]
```diff
@@ -45,6 +48,74 @@ public void testCreateVoidDDMFormField() throws Exception {
...
+			ddmFormFieldValidation.getDDMFormFieldValidationExpression();
+
+		Assert.assertEquals(
+			"expression", ddmFormFieldValidationExpression.getValue());
+
+		Assert.assertNull(ddmFormFieldValidationExpression.getName());
+
+		Assert.assertNull(ddmFormFieldValidation.getParameterLocalizedValue());
+	}
+
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/util/DDMFormFieldFactoryHelperTest.java:118]: Nice job, @rodrigopaulino! :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormInstanceSettings.java [hunk 2]
```diff
@@ -139,6 +139,15 @@
 	)
 	public String emailToAddress();
 
+	@DDMFormField(
+		label = "%expiration-date", type = "date",
+		validationErrorMessage = "%please-enter-a-valid-expiration-date-only-future-dates-are-accepted",
+		validationExpression = "futureDates(expirationDate, \"{parameter}\")",
+		validationExpressionName = "futureDates",
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormInstanceSettings.java:146]: @rodrigopaulino, can you please give more details about the purpose of the parameter `validationExpressionName`?

Thanks!

Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormInstanceSettings.java:146]: Yeah, of course!

This is necessary for the same reason DDMValidations have [names](https://github.com/liferay/liferay-portal/blob/c5a2f60bc87ff963fe1b71b63a0519e1ae913cad/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/validation/DDMValidation.java#L26) and so DDMFormFieldValidationExpression can replace the parameter accordingly ([here](https://github.com/liferay/liferay-portal/blob/c5a2f60bc87ff963fe1b71b63a0519e1ae913cad/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormFieldValidationExpression.java#L76)).

Hope this clarifies.

Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/model/DDMFormInstanceSettings.java:146]: Thanks for your clarification, pal! :wink:

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**